### PR TITLE
Enhance discord integration with custom rendered pictures

### LINKED ISSUE #57: Send rendered pictures to discord
The idea is that players are able to grab more information without having to open the website just by looking at the discord. This is achieved by sending a nice looking picture alongside the announcements by the discord bot.

As of right now, these are all announcements made on the discord:
- Epochtal week started
- Submissions to the leaderboard
- Leaderboard locked at end of week
- Demo reports summary at end of week
- Failed routine calls

Here's the change I propose for "Epochtal week started":
When the week starts, send a picture identical to the "This Week's Map" display on the website, alongside the message. This makes it much easier to decipher which map was picked, due to the added thumbnail as well as the clearly separated by line title and author. Additionally one can easily see how favorable a map was through up/downvotes. Additional information can always be added, lmk if you have an idea what to put here.

Here's the change I propose for "Submissions to the leaderboard":
After each submission, the run just above and below should be shown in a leaderboard-style just like on the website, with additional indicators marking that another player has lost a place and got moved downwards. Perhaps even show two runs below on placement change, I will have to tinker around with what looks nice. Also perhaps, a line indicating how many places the submitting player moved *up* would be neat.
For non-scored categories, simply posting the run with it's comment in leaderboard-style should be plenty, though I'm open for suggestions.

As for "Leaderboard locked at end of week", I'm not sure what to do here.
One idea is to post the top 3 or top 5 players in a picture, as with discord stupid formatting, showing the entire leaderboard or generally just a dynamic amount, risks getting put into this silly frame where half the picture is not visible. Since epochtal does not have a ""main"" category, how should we determine which categories are posted on the discord? Multiple pictures sent in multiple messages for all scored categories? Or just the very first scored category in the list of categories?

"Failed routine calls" are in my opinion already formatted as neatly as they can be.

"Demo reports summary" is probably too complex to generate pictures for and isn't particularly hard to read anyways, so I would leave this one untouched as it is.

Maybe-potentially-out-of-scope-but-cool would be the addition of new messages. The stream announcement could be sent through the bot as well. I'm thinking maybe adding a neat picture containing a short summary of the *amount* of runs in each category that will be shown on stream would be cool? I would add this to the leaderboard locked message, however I feel that one will become to cluttered. Additionally, knowing what's shown on stream sounds more like the fitting place for this.

Lots of room for discussion, leave thoughts plz!!

P.S: is this frontend, technically?

### PR DESCRIPTION
**Pull request by PancakeTAS**
This PR updates the discord util to automatically generate pictures and send them alongside the discord message.

Big work in progress, check smaller updates in the discord.

Closes #57

### CODE REVIEW

File: util/discord.js [hunk 1]
```diff
@@ -30,6 +40,39 @@ module.exports = async function (args, context = epochtal) {
...
+      const image = await render(
+        buffer,
+        context.data.week.map.title,
+        context.data.week.map.author,
+        context.data.week.map.upvotes,
+        context.data.week.map.downvotes
+      );
+
+      const imagePath = (await tmppath()) + ".png";
+      Bun.write(imagePath, image);
```
Comment (PolarizedIons) [inline util/discord.js:64]: does this ever get cleaned up?

Comment (p2r3) [reply util/discord.js:64]: (also that's an async write without an await!!)

Comment (PancakeTAS) [reply util/discord.js:64]: how about 7 lines below?

Comment (PolarizedIons) [reply util/discord.js:64]: nvm, TIL `unlink` isn't just for symlinks

Comment (PancakeTAS) [reply util/discord.js:64]: > (also that's an async write without an await!!)
this is why we need eslint, lmfao

Comment (soni801) [reply util/discord.js:64]: nah thats just an excuse to write bad code :Kappa:

**Comment by PancakeTAS** [comment]
not going to work on this anymore

**Pull request closed without merging**